### PR TITLE
1419576: Pre-date certs to ease clock skew issues

### PR DESCRIPTION
**Pull request by wottop**

### CODE REVIEW

File: server/src/test/java/org/candlepin/service/impl/DefaultEntitlementCertServiceAdapterTest.java [hunk 1]
```diff
@@ -367,6 +368,38 @@ public void temporaryCertificateForUnmappedGuests() throws Exception {
...
+        // pool start date is more than an hour ago, use it
+        Calendar cal = Calendar.getInstance();
+        cal.add(Calendar.HOUR, -2);
+        pool.setStartDate(cal.getTime());
+        X509Certificate result = certServiceAdapter.createX509Certificate(entitlement,
+            product, new HashSet<Product>(),
+            getProductModels(product, new HashSet<Product>(), "prefix", entitlement),
+            new BigInteger("1234"), keyPair, true);
+        // cert does not capture the milliseconds. truncating.
+        assert (result.getNotBefore().getTime() == pool.getStartDate().getTime() / 1000 * 1000);
```
Comment (Ceiu) [inline server/src/test/java/org/candlepin/service/impl/DefaultEntitlementCertServiceAdapterTest.java]: Use assertTrue or assertEquals here

File: server/src/test/java/org/candlepin/service/impl/DefaultEntitlementCertServiceAdapterTest.java [hunk 2]
```diff
@@ -367,6 +368,38 @@ public void temporaryCertificateForUnmappedGuests() throws Exception {
...
+        // pool start date is less than an hour ago, so an hour gets used
+        cal.add(Calendar.MINUTE, 90);
+        pool.setStartDate(cal.getTime());
+        result = certServiceAdapter.createX509Certificate(entitlement,
+                product, new HashSet<Product>(),
+                getProductModels(product, new HashSet<Product>(), "prefix", entitlement),
+                new BigInteger("1234"), keyPair, true);
+        assert (result.getNotBefore().getTime() < pool.getStartDate().getTime() / 1000 * 1000);
```
Comment (Ceiu) [inline server/src/test/java/org/candlepin/service/impl/DefaultEntitlementCertServiceAdapterTest.java]: assertTrue here

File: server/src/main/java/org/candlepin/service/impl/DefaultEntitlementCertServiceAdapter.java [hunk 3]
```diff
@@ -171,10 +172,13 @@ public X509Certificate createX509Certificate(Entitlement ent, Product product, S
             extensions = prepareV1Extensions(products, ent, contentPrefix,
                 promotedContent);
         }
-
         setupEntitlementEndDate(ent);
+        Calendar cal = Calendar.getInstance();
+        cal.add(Calendar.HOUR, -1);
+        Date startDate = ent.getStartDate().getTime() < cal.getTime().getTime() ?
```
Comment (awood) [inline server/src/main/java/org/candlepin/service/impl/DefaultEntitlementCertServiceAdapter.java]: You might want to use `Date`'s `before` function here.

```
(ent.getStartDate().before(cal.getTime()) ? ... : ...
```

**Comment by wottop** [comment]
buildbot: retest this please

**Pull request merged**